html/modules/custom/reliefweb_rivers/src/Services/TopicRiver.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace Drupal\reliefweb_rivers\Services;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Pager\PagerManagerInterface;
+use Drupal\Core\Pager\PagerParametersInterface;
+use Drupal\Core\StringTranslation\TranslationInterface;
+use Drupal\Core\Url;
+use Drupal\reliefweb_api\Services\ReliefWebApiClient;
+use Drupal\reliefweb_rivers\RiverServiceBase;
+use Drupal\reliefweb_utility\Helpers\HtmlSummarizer;
+
+/**
+ * Service class to retrieve job resource for the job rivers.
+ */
+class TopicRiver extends RiverServiceBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $river = 'topics';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $resource = 'topics';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $entityType = 'node';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $bundle = 'topic';
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPageTitle() {
+    return $this->t('Topics');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getViews() {
+    return [
+      'all' => $this->t('All Topics'),
+    ];
+  }
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Pager\PagerManagerInterface $pager_manager
+   *   The pager manager service.
+   * @param \Drupal\Core\Pager\PagerParametersInterface $pager_parameters
+   *   The pager parameter service.
+   * @param \Drupal\reliefweb_api\Services\ReliefWebApiClient $api_client
+   *   The ReliefWeb API Client service.
+   * @param \Drupal\Core\StringTranslation\TranslationInterface $string_translation
+   *   The translation manager service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   */
+  public function __construct(PagerManagerInterface $pager_manager, PagerParametersInterface $pager_parameters, ReliefWebApiClient $api_client, TranslationInterface $string_translation, EntityTypeManagerInterface $entity_type_manager) {
+    $this->pagerManager = $pager_manager;
+    $this->pagerParameters = $pager_parameters;
+    $this->apiClient = $api_client;
+    $this->stringTranslation = $string_translation;
+    $this->url = static::getRiverUrl($this->bundle);
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRiverContent() {
```

Comment:
The topics landing page doesn't use a pager because it's a relatively small, curated  list of topics so the query to get the totalCount etc. can be removed.